src/traces/box/hover.js

```diff
@@ -14,94 +14,167 @@ var Fx = require('../../components/fx');
-    else pointData.color = trace.fillcolor;
+            pointData[posLetter + '0'] = posAxis.c2p(di.pos + t.bPos - t.bdPos, true);
+            pointData[posLetter + '1'] = posAxis.c2p(di.pos + t.bPos + t.bdPos, true);
 
-    pointData[posLetter + '0'] = posAxis.c2p(di.pos + t.bPos - t.bdPos, true);
-    pointData[posLetter + '1'] = posAxis.c2p(di.pos + t.bPos + t.bdPos, true);
+            Axes.tickText(posAxis, posAxis.c2l(di.pos), 'hover').text;
+            pointData[posLetter + 'LabelVal'] = di.pos;
 
-    Axes.tickText(posAxis, posAxis.c2l(di.pos), 'hover').text;
-    pointData[posLetter + 'LabelVal'] = di.pos;
+            // box plots: each "point" gets many labels
+            var usedVals = {};
+            var attrs = ['med', 'min', 'q1', 'q3', 'max'];
 
-    // box plots: each "point" gets many labels
-    var usedVals = {},
-        attrs = ['med', 'min', 'q1', 'q3', 'max'],
-        attr,
-        pointData2;
-    if(trace.boxmean) attrs.push('mean');
-    if(trace.boxpoints) [].push.apply(attrs, ['lf', 'uf']);
+            if(trace.boxmean) attrs.push('mean');
+            if(trace.boxpoints) [].push.apply(attrs, ['lf', 'uf']);
 
-    for(var i = 0; i < attrs.length; i++) {
-        attr = attrs[i];
+            for(i = 0; i < attrs.length; i++) {
+                var attr = attrs[i];
 
-        if(!(attr in di) || (di[attr] in usedVals)) continue;
-        usedVals[di[attr]] = true;
+                if(!(attr in di) || (di[attr] in usedVals)) continue;
+                usedVals[di[attr]] = true;
 
-        // copy out to a new object for each value to label
-        val = valAxis.c2p(di[attr], true);
-        pointData2 = Lib.extendFlat({}, pointData);
-        pointData2[valLetter + '0'] = pointData2[valLetter + '1'] = val;
-        pointData2[valLetter + 'LabelVal'] = di[attr];
-        pointData2.attr = attr;
+                // copy out to a new object for each value to label
+                var val = valAxis.c2p(di[attr], true);
+                pointData2 = Lib.extendFlat({}, pointData);
+                pointData2[valLetter + '0'] = pointData2[valLetter + '1'] = val;
+                pointData2[valLetter + 'LabelVal'] = di[attr];
+                pointData2.attr = attr;
 
-        if(attr === 'mean' && ('sd' in di) && trace.boxmean === 'sd') {
-            pointData2[valLetter + 'err'] = di.sd;
+                if(attr === 'mean' && ('sd' in di) && trace.boxmean === 'sd') {
+                    pointData2[valLetter + 'err'] = di.sd;
+                }
+                // only keep name on the first item (median)
+                pointData.name = '';
+
+                closeData.push(pointData2);
+            }
         }
-        pointData.name = ''; // only keep name on the first item (median)
-        closeData.push(pointData2);
     }
+
+    if(hoveron.indexOf('points') !== -1) {
+        var xPx = xa.c2p(xval);
+        var yPx = ya.c2p(yval);
+
+        // do not take jitter into consideration in compare hover modes
```

Comment:
Anyone disagree here?

Reply:
I agree, and might even take it farther - can we make all of the hover labels line up horizontally, even though the points do not?

Reply:
oh hmm I guess https://github.com/plotly/plotly.js/pull/2094/files#r145223324 supersedes this comment, except perhaps if we show a point and stats in compare mode.